src/offset/local/unix.rs

```diff
@@ -37,35 +37,89 @@ pub(super) fn naive_to_local(d: &NaiveDateTime, local: bool) -> LocalResult<Date
 }
 
 fn offset(d: NaiveDateTime, local: bool) -> LocalResult<FixedOffset> {
-    let info = unsafe {
-        INIT.call_once(|| {
-            INFO = Some(TimeZone::local().expect("unable to parse localtime info"));
-        });
-        INFO.as_ref().unwrap()
-    };
+    TZINFO.with(|cache| {
+        let needs_refresh = match &*cache.borrow() {
+            Some(existing) if existing.source.out_of_date() => true,
+            None => true,
+            _ => false,
+        };
+        if needs_refresh {
+            *cache.borrow_mut() = Some(Cache::new());
+        }
 
-    if local {
-        // we pass through the year as the year of a local point in time must either be valid in that locale, or
-        // the entire time was skipped in which case we will return LocalResult::None anywa.
-        match info
-            .find_local_time_type_from_local(d.timestamp(), d.year())
-            .expect("unable to select local time type")
-        {
-            LocalResult::None => LocalResult::None,
-            LocalResult::Ambiguous(early, late) => LocalResult::Ambiguous(
-                FixedOffset::east(early.offset()),
-                FixedOffset::east(late.offset()),
+        if local {
+            // we pass through the year as the year of a local point in time must either be valid in that locale, or
+            // the entire time was skipped in which case we will return LocalResult::None anywa.
+            match cache
+                .borrow()
+                .as_ref()
+                .unwrap()
+                .zone
+                .find_local_time_type_from_local(d.timestamp(), d.year())
+                .expect("unable to select local time type")
+            {
+                LocalResult::None => LocalResult::None,
+                LocalResult::Ambiguous(early, late) => LocalResult::Ambiguous(
+                    FixedOffset::east(early.offset()),
+                    FixedOffset::east(late.offset()),
+                ),
+                LocalResult::Single(tt) => LocalResult::Single(FixedOffset::east(tt.offset())),
+            }
+        } else {
+            LocalResult::Single(FixedOffset::east(
+                cache
+                    .borrow()
+                    .as_ref()
+                    .unwrap()
+                    .zone
+                    .find_local_time_type(d.timestamp())
+                    .expect("unable to select local time type")
+                    .offset(),
+            ))
+        }
+    })
+}
+
+thread_local! {
+    static TZINFO: RefCell<Option<Cache>> = RefCell::new(None);
```

Comment:
Style nit: I think calling this `TZ_INFO` would be more idiomatic.

So we need the `Option` here because we can't initialize a real `Cache` in a `static` context? Might be worth a comment.